go/pkg/client/cas.go

```diff
@@ -1054,6 +1055,10 @@ func (c *Client) downloadSingle(ctx context.Context, dg digest.Digest, reqs map[
 	}
 	r := rs[0]
 	rs = rs[1:]
```

Comment:
Can you add a comment as to what you're checking for here? I assume making sure there is only one requestor for the blob so we can use their context?

Reply:
Yep, done!